core/src/main/scala/org/broadinstitute/dsde/rawls/monitor/WorkspaceBillingAccountActor.scala

```diff
@@ -0,0 +1,275 @@
+package org.broadinstitute.dsde.rawls.monitor
+
+import akka.actor.typed.Behavior
+import akka.actor.typed.scaladsl.Behaviors
+import cats.Applicative
+import cats.effect.IO
+import cats.effect.unsafe.implicits.global
+import cats.implicits._
+import com.google.api.services.cloudbilling.model.ProjectBillingInfo
+import com.typesafe.scalalogging.LazyLogging
+import org.broadinstitute.dsde.rawls.RawlsException
+import org.broadinstitute.dsde.rawls.dataaccess.slick.{BillingAccountChange, WriteAction}
+import org.broadinstitute.dsde.rawls.dataaccess.{GoogleServicesDAO, SlickDataSource}
+import org.broadinstitute.dsde.rawls.model._
+import org.broadinstitute.dsde.rawls.monitor.migration.MigrationUtils.Implicits._
+import org.broadinstitute.dsde.rawls.monitor.migration.MigrationUtils.Outcome
+import org.broadinstitute.dsde.rawls.monitor.migration.MigrationUtils.Outcome.{Failure, Success}
+
+import java.time.Instant
+import scala.concurrent.duration._
+
+object WorkspaceBillingAccountActor {
+  sealed trait WorkspaceBillingAccountsMessage
+
+  case object UpdateBillingAccounts extends WorkspaceBillingAccountsMessage
+
+  def apply(dataSource: SlickDataSource,
+            gcsDAO: GoogleServicesDAO,
+            initialDelay: FiniteDuration,
+            pollInterval: FiniteDuration): Behavior[WorkspaceBillingAccountsMessage] =
+    Behaviors.setup { context =>
+      val actor = WorkspaceBillingAccountActor(dataSource, gcsDAO)
+      Behaviors.withTimers { scheduler =>
+        scheduler.startTimerAtFixedRate(UpdateBillingAccounts, initialDelay, pollInterval)
+        Behaviors.receiveMessage {
+          case UpdateBillingAccounts =>
+            try actor.updateBillingAccounts.unsafeRunSync()
+            catch {
+              case t: Throwable => context.executionContext.reportFailure(t)
+            }
+            Behaviors.same
+        }
+      }
+    }
+}
+
+final case class WorkspaceBillingAccountActor(dataSource: SlickDataSource, gcsDAO: GoogleServicesDAO)
+  extends LazyLogging {
+
+  import dataSource.dataAccess._
+  import dataSource.dataAccess.driver.api._
+
+  /* Sync billing account changes to billing projects and their associated workspaces with Google
```

Comment:
I probably don't have all the context -- but what does a "billing account change" represent, and how is the table populated in Rawls?

Reply:
So when you update which google billing account a rawls billing project uses, rawls does this
```haskell
do
    project <- loadBillingProject projectName
    when (project.billingAcount != newBillingAccount) $ do
        setBillingAccount projectName newBillingAccount
        createChangeRecord projectName project.billingAccount
```